binderhub/repoproviders.py

```diff
@@ -117,16 +117,27 @@ def _default_auth(self):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        spec_parts = self.spec.split('/')
+        self.user, self.repo, self.unresolved_ref = self.process_spec(self.spec)
+        self.repo = strip_suffix(self.repo, ".git")
+
+    @staticmethod
+    def process_spec(spec):
+        """Tokenizes a Github Spec into parts, error if spec invalid."""
```

Comment:
-> "Tokenize a GitHub spec into user, repo, ref."? Mostly I am a sucker for "Tokenize" over "Tokenizes" (https://www.python.org/dev/peps/pep-0257/#specification)

Reply:
Ah, good catch